File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

// declaring functions
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Enter Text: ");

    float words = count_words(text);
    float sentences = count_sentences(text);
    float letters = count_letters(text);

    printf("Words: %f, sentences: %f, letters: %f\n", words, sentences, letters);

    // gets number of letters, sentences, words

    float L = (letters / (float)words) * 100;
    float S = (sentences / (float)words) * 100;
    printf("L: %f, S: %f\n", L, S);

    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int count_letters(string text)
{
    // count letters in for loop
    int letters = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (islower(text[i]) | isupper(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{
    // count words in for loop
    int words = 0;

    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
    }
    return words + 1;
}

int count_sentences(string text)
{
    // count sentences in for loop
    int sentences = 0;

    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == '!' || text[i] == '?' || text[i] == '.')
        {
            sentences++;
        }
    }
    return sentences;
}

File: sentimental-readability/readability.py
# TODO
from cs50 import get_string


def count_letters(text):
    letters = 0
    for letter in text.lower():
        if letter.islower():
            letters += 1

    return letters


def count_words(text):
    words = 0
    for letter in text:
        if (letter == ' '):
            words += 1

    return words + 1


def count_sentences(text):
    sentences = 0
    for letter in text:
        if letter == '!' or letter == '?' or letter == '.':
            sentences += 1

    return sentences


def main():

    text = get_string("Text: ")
    words = count_words(text)
    sentences = count_sentences(text)
    letters = count_letters(text)

    # gets number of letters, sentences, words
    L = (letters / words) * 100
    S = (sentences / words) * 100

    index = round(0.0588 * L - 0.296 * S - 15.8)
    if index >= 16:
        print("Grade 16+\n")
    elif index < 1:
        print("Before Grade 1\n")
    else:
        print(f"Grade {index}\n")


main()
